format/avro/decoders/logicaltypes.go

```diff
@@ -0,0 +1,105 @@
+package decoders
+
+import (
+	"errors"
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/scalar"
+	"time"
+)
+
+type Precision int
+
+const (
+	SECOND = iota
+	MILLISECOND
+	MICROSECOND
+	NANOSECOND
+)
+
+func logicalMapperForSchema(schema schema.SimplifiedSchema) scalar.Mapper {
+	switch schema.LogicalType {
+	case "timestamp":
+		return TimestampMapper{Precision: SECOND}
+	case "timestamp-millis":
+		return TimestampMapper{Precision: MILLISECOND}
+	case "timestamp-micros":
+		return TimestampMapper{Precision: MICROSECOND}
+	case "timestamp-nanos":
+		return TimestampMapper{Precision: NANOSECOND}
+	case "time":
+		return TimeMapper{Precision: SECOND}
+	case "time-millis":
+		return TimeMapper{Precision: MILLISECOND}
+	case "time-micros":
+		return TimeMapper{Precision: MICROSECOND}
+	case "time-nanos":
+		return TimeMapper{Precision: NANOSECOND}
+	case "date":
+		return DateMapper{}
+	default:
+		return nil
+	}
+}
+
+type TimestampMapper struct {
+	Precision Precision
+}
+
+func (t TimestampMapper) MapScalar(s scalar.S) (scalar.S, error) {
+	v, ok := s.Actual.(int64)
```

Comment:
If you know it must be a `int64` unless there is a bug you can use`s.ActualS()` which will panic

Reply:
Nice! swapped